.github/workflows/ci.yml

```diff
@@ -19,87 +20,171 @@ jobs:
-        - platform: { name: Android, os: ubuntu-22.04  }
-          config: { name: x86_64, flags: -GNinja -DCMAKE_ANDROID_ARCH_ABI=x86_64 -DCMAKE_SYSTEM_NAME=Android -DSFML_BUILD_TEST_SUITE=FALSE -DCMAKE_TOOLCHAIN_FILE=$GITHUB_WORKSPACE/android-ndk-r23b/build/cmake/android.toolchain.cmake -DCMAKE_ANDROID_NDK=$GITHUB_WORKSPACE/android-ndk-r23b -DCMAKE_ANDROID_NDK_TOOLCHAIN_VERSION=clang -DCMAKE_ANDROID_STL_TYPE=c++_shared -DCMAKE_ANDROID_API=26 -DANDROID_PLATFORM=26 }
-        - platform: { name: Linux GCC, os: ubuntu-22.04  }
-          config: { name: Static DRM, flags: -GNinja -DSFML_USE_DRM=TRUE -DSFML_RUN_DISPLAY_TESTS=FALSE }
-        - platform: { name: Linux GCC, os: ubuntu-22.04  }
-          config: { name: Shared DRM, flags: -GNinja -DBUILD_SHARED_LIBS=TRUE -DSFML_USE_DRM=TRUE -DSFML_RUN_DISPLAY_TESTS=FALSE }
-        - platform: { name: macOS , os: macos-12 }
-          config: { name: System Deps, flags: -GNinja -DBUILD_SHARED_LIBS=TRUE -DSFML_USE_SYSTEM_DEPS=TRUE }
+          config: { name: arm64-v8a }
+        - platform: { name: Android, os: ubuntu-22.04 }
+          config: { name: x86_64 }
+        - platform: { name: Linux GCC DRM, os: ubuntu-22.04 }
+          config: { name: Shared }
+        - platform: { name: Linux GCC DRM, os: ubuntu-22.04 }
+          config: { name: Static }
+
     steps:
     - name: Checkout Code
       uses: actions/checkout@v3
 
+    - name: Set Visual Studio Architecture
+      if: contains(matrix.platform.name, 'Windows VS')
+      uses: ilammy/msvc-dev-cmd@v1
+      with:
+        arch: ${{ contains(matrix.platform.name, 'x86') && 'x86' || 'x64' }}
+
+    - name: Set Compiler
+      shell: bash
+      run: |
+        if [[ "${{ matrix.platform.name }}" == *"ClangCL"* ]]; then
+          INSTALLATION_PATH=$(vswhere -latest -requires Microsoft.VisualStudio.Component.VC.Llvm.Clang -property installationPath)
+          CLANGCL_PATH="${INSTALLATION_PATH//\\//}/VC/Tools/Llvm/x64/bin/clang-cl.exe"
+          echo "CC=\"$CLANGCL_PATH\"" >> $GITHUB_ENV
+          echo "CXX=\"$CLANGCL_PATH\"" >> $GITHUB_ENV
+        elif [[ "${{ matrix.platform.name }}" == *"Clang"* ]]; then
+          echo "CC=clang" >> $GITHUB_ENV
+          echo "CXX=clang++" >> $GITHUB_ENV
+        elif [[ "${{ matrix.platform.name }}" == *"MinGW"* ]]; then
+          echo "CC=gcc" >> $GITHUB_ENV
+          echo "CXX=g++" >> $GITHUB_ENV
+        elif [[ "${{ matrix.platform.name }}" == *"GCC"* ]]; then
+          echo "CC=gcc" >> $GITHUB_ENV
+          echo "CXX=g++" >> $GITHUB_ENV
+        elif [[ "${{ matrix.platform.name }}" == *"Intel"* ]]; then
+          echo "CC=icx" >> $GITHUB_ENV
+          echo "CXX=icpx" >> $GITHUB_ENV
+        fi
+
+    - name: Get Latest CMake and Ninja
+      uses: lukka/get-cmake@latest
+      with:
+        cmakeVersion: latest
+        ninjaVersion: latest
```

Comment:
Question for CMake is, whether we should go with the latest at all times, or whether we should actually stick to the minimum required version.

Reply:
`CMAKE_MSVC_DEBUG_INFORMATION_FORMAT` was only introduced in CMake 3.25 and isn't really relevant for most users who don't intend to use `ccache`. It speeds up workflows by quite a bit but wouldn't be a reason for me to bump up the minimum required version. Since CMake configuration changes are very infrequent and mostly very conservative I don't think it is really necessary to test the minimum version requirement as part of CI. If we see CMake changes in a patch that are concerning we would already check compatibility there.